token-metadata/program/src/processor/metadata/print.rs

```diff
@@ -0,0 +1,194 @@
+    // let mint_authority_info = next_account_info(account_info_iter)?;
+    // let payer_account_info = next_account_info(account_info_iter)?;
+    // let owner_account_info = next_account_info(account_info_iter)?;
+    // let token_account_info = next_account_info(account_info_iter)?;
+    // let update_authority_info = next_account_info(account_info_iter)?;
+    // let master_metadata_account_info = next_account_info(account_info_iter)?;
+    // let token_program_account_info = next_account_info(account_info_iter)?;
+    // let system_account_info = next_account_info(account_info_iter)?;
+
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_metadata_info = ctx.accounts.edition_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_account_info = ctx.accounts.edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_mint_info = ctx.accounts.edition_mint_info;
+    let edition_token_account_owner_info = ctx.accounts.edition_token_account_owner_info;
+    let edition_token_account_info = ctx.accounts.edition_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_mint_authority_info = ctx.accounts.edition_mint_authority_info;
+    let edition_token_record_info = ctx.accounts.edition_token_record_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_edition_info = ctx.accounts.master_edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_marker_pda_info = ctx.accounts.edition_marker_pda_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let payer_info = ctx.accounts.payer_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_owner_info = ctx.accounts.master_token_account_owner_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_info = ctx.accounts.master_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_metadata_info = ctx.accounts.master_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let update_authority_info = ctx.accounts.update_authority_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let token_program = ctx.accounts.token_program_info;
+    let ata_program = ctx.accounts.ata_program_info;
+    let sysvar_instructions = ctx.accounts.sysvar_instructions_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let system_program = ctx.accounts.system_program_info;
```

Comment:
It seems like all the other program IDs are explicitly checked but I don't see where this one is explicitly checked in `process_mint_new_edition_from_master_edition_via_token_logic` or elsewhere.  I'm pretty sure many things don't work if you pass in the wrong system program but might be more debuggable if there was an explicit check like with other programs.